### PR TITLE
Render control plane resources using giantswarm/cluster

### PR DESCRIPTION
**Pull request by glitchcrab**
towards https://github.com/giantswarm/giantswarm/issues/31698

This PR:

- renders the `kubeadmcontrolplane` using `giantswarm/cluster`.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [ ] Update `/examples` if required.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,52 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+    with(select(.global.controlPlane.diskSizeGB != null); .global.controlPlane.machineTemplate.diskSizeGB = .global.controlPlane.diskSizeGB) |
+    with(select(.global.controlPlane.placementPolicy != null); .global.controlPlane.machineTemplate.placementPolicy = .global.controlPlane.placementPolicy) |
+    with(select(.global.controlPlane.sizingPolicy != null); .global.controlPlane.machineTemplate.sizingPolicy = .global.controlPlane.sizingPolicy) |
+    with(select(.global.controlPlane.storageProfile != null); .global.controlPlane.machineTemplate.storageProfile = .global.controlPlane.storageProfile) |
+    with(select(.global.controlPlane.template != null); .global.controlPlane.machineTemplate.template = .global.controlPlane.template)' values.yaml
+```
+
+## Cleanup
+
+Final tidyup to remove deprecated values:
```
Comment (vxav) [inline CHANGELOG.md:36]: Just my opinion but I found it better with one single command to process the values

File: helm/cluster-cloud-director/values.schema.json [hunk 2]
```diff
@@ -800,18 +800,11 @@
                     "type": "object",
                     "title": "Control plane",
                     "required": [
-                        "catalog",
-                        "template",
```
Comment (vxav) [inline helm/cluster-cloud-director/values.schema.json:804]: 👌

File: helm/cluster-cloud-director/templates/helpers/_workers.tpl [hunk 3]
```diff
@@ -3,11 +3,13 @@ Generates template spec for worker machines.
 */}}
 {{- define "worker-vcdmachinetemplate-spec" -}}
 {{- $pool := $.nodePool.config | deepCopy -}}
+{{- $pool = set $pool "diskSize" ( include "calculateDiskBytes" $pool.diskSizeGB ) -}}
+{{- $pool = unset $pool "diskSizeGB" -}}
```
Comment (glitchcrab) [inline helm/cluster-cloud-director/templates/helpers/_workers.tpl:7]: this was missed in the previous cluster integration PR (to render worker resources)

Comment (vxav) [reply helm/cluster-cloud-director/templates/helpers/_workers.tpl:7]: Where does that function come from?

Comment (glitchcrab) [reply helm/cluster-cloud-director/templates/helpers/_workers.tpl:7]: thanks for pointing out that i completely forgot to commit two files :facepalm: 

https://github.com/giantswarm/cluster-cloud-director/pull/374/files#diff-84ed6144d005d761e40cefbbcc342f5b12576a808d0cf6701bfe03cac8186233

File: helm/cluster-cloud-director/templates/helpers/_workers.tpl [hunk 4]
```diff
@@ -3,11 +3,15 @@ Generates template spec for worker machines.
...
 {{- $pool = unset $pool "replicas" -}}
 {{- $pool = unset $pool "machineHealthCheck" -}}
+{{- $pool = unset $pool "customNodeLabels" -}}
+{{- $pool = unset $pool "customNodeTaints" -}}
```
Comment (glitchcrab) [inline helm/cluster-cloud-director/templates/helpers/_workers.tpl:11]: these two aren't usually set but if they are then they leak into the vcdmachinetemplate spec so we need to delete them (does not affect the control plane)

File: CHANGELOG.md (hunk 1 above)
Comment (vxav) [reply CHANGELOG.md:116]: It makes sense now that you've added the other migration cmds which depends on the values used.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,145 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+## OIDC config
+
+`caFile` has been renamed to `caPem`.
+
+```
+yq eval --inplace 'with(select(.global.controlPlane.oidc.caFile != null); .global.controlPlane.oidc.caPem = .global.controlPlane.oidc.caFile)' values.yaml
+```
+
+Additionally, `groupsPrefix` and `usernamePrefix` are removed (see the **Cleanup** section below).
```
Comment (vxav) [inline CHANGELOG.md:83]: Our customer uses them though 
https://github.com/giantswarm/telekom-panamax/blob/main/bases/cluster/values.yaml#L66-L67

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,145 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+If provided, NTP server addresses need to be migrated to the new location.
+
+```
+yq eval --inplace 'with(select(.global.connectivity.ntp.servers != null); .cluster.providerIntegration.components.systemd.timesyncd.ntp = .global.connectivity.ntp.servers)' values.yaml
+```
+
+## Upstream proxy settings
+
+Upstream proxy configuration is no longer read from the `.global.connectivity.proxy.secretName` value so this must be removed (see the **Cleanup** section below).
```
Comment (vxav) [inline CHANGELOG.md:103]:
```suggestion
Upstream proxy configuration is no longer read from the `.global.connectivity.proxy.secretName` value so this must be removed (see the **Cleanup** section below).

Additionally the `.global.connectivity.proxy.httpProxy`, `.global.connectivity.proxy.httpsProxy` and `.global.connectivity.proxy.noProxy` values should be populated.
```

Comment (glitchcrab) [reply CHANGELOG.md:103]: This was covered in the upgrade guide for the previous release

File: CHANGELOG.md (hunk 5 above)
Comment (glitchcrab) [reply CHANGELOG.md:83]: https://github.com/giantswarm/cluster/pull/363

**Review by vxav — APPROVED** [review]
I haven't tested this myself but LGTM if confirmed to work (especially ignition).
Couple CI failures due to schema switcheroo though

**Comment by glitchcrab** [comment]
i'm going to go ahead and merge this - everything comes up as expected:

```
shw cluster-cloud-director ☸ wealdy-test-2-admin@wealdy-test-2:  cluster/render-control-plane [$?] ✗  kg no
NAME         STATUS   ROLES           AGE     VERSION
wealdy183e   Ready    control-plane   2m17s   v1.27.14
wealdy1d2c   Ready    <none>          67s     v1.27.14
wealdy4404   Ready    <none>          58s     v1.27.14
wealdya2ed   Ready    <none>          46s     v1.27.14
shw cluster-cloud-director ☸ wealdy-test-2-admin@wealdy-test-2:  cluster/render-control-plane [$?] ➜  kg po -A
NAMESPACE     NAME                                        READY   STATUS    RESTARTS   AGE
giantswarm    chart-operator-68bb8d9498-2zv7k             1/1     Running   0          2m17s
kube-system   cilium-456xg                                1/1     Running   0          68s
kube-system   cilium-9hdn9                                1/1     Running   0          47s
kube-system   cilium-n5jwh                                1/1     Running   0          112s
kube-system   cilium-operator-54f7ccf6ff-hdmwn            1/1     Running   0          112s
kube-system   cilium-operator-54f7ccf6ff-mqwcf            1/1     Running   0          112s
kube-system   cilium-qbmc8                                1/1     Running   0          59s
kube-system   coredns-controlplane-57d9d5b964-r5674       1/1     Running   0          87s
kube-system   coredns-workers-68c57ff486-lwwmb            1/1     Running   0          87s
kube-system   coredns-workers-68c57ff486-xmjcm            1/1     Running   0          72s
kube-system   csi-vcd-controllerplugin-0                  3/3     Running   0          88s
kube-system   csi-vcd-nodeplugin-2vwqr                    2/2     Running   0          14s
kube-system   csi-vcd-nodeplugin-4hr94                    2/2     Running   0          38s
kube-system   csi-vcd-nodeplugin-4vvcf                    2/2     Running   0          34s
kube-system   etcd-wealdy183e                             1/1     Running   0          2m17s
kube-system   hubble-relay-57c47f8c78-fx7bc               1/1     Running   0          112s
kube-system   hubble-ui-7488cd6d9b-gzjx6                  2/2     Running   0          112s
kube-system   kube-apiserver-wealdy183e                   1/1     Running   0          2m17s
kube-system   kube-controller-manager-wealdy183e          1/1     Running   0          2m17s
kube-system   kube-scheduler-wealdy183e                   1/1     Running   0          2m18s
kube-system   vmware-cloud-director-ccm-d7ff6b8db-qsh6f   1/1     Running   0          88s
```

**Comment by glitchcrab** [comment]
static routes service on the control plane is fine:

```
wealdy183e ~ # cat /etc/systemd/system/static-routes.service
[Unit]
Description=Add static network routes.
After=set-networkd-units.service
After=systemd-networkd.service
BindsTo=systemd-networkd.service
[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/bash -cv 'sleep 3'
ExecStart=/usr/bin/bash -cv 'ip route add 193.16.42.72 via 10.205.105.1'
[Install]
WantedBy=multi-user.target
```

File: CHANGELOG.md (hunk 5 above)
Comment (glitchcrab) [reply CHANGELOG.md]: this is merged; the options are available again

**Pull request merged**